### PR TITLE
Correct HTTP path to `conda_env` test data and remove empty `conda.env.cli` submodule

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Followup to #13168

Missed updating a few HTTP urls pointing to the old `tests/conda_env` path.

And remove empty `conda.env.cli` submodule.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by jezdez** [comment]
See #13458 😬

**Comment by kenodegard** [comment]
just saw it!

**Comment by jezdez** [comment]
Close mine since you found one more instance

**Comment by jezdez** [comment]
D'oh!

**Comment by jezdez** [comment]
Let's reopen yours

**Comment by jezdez** [comment]
Let's reopen this

**Comment by kenodegard** [comment]
🤦🏼‍♂️ I was too quick

File: conda/env/cli/__init__.py
Comment (kenodegard) [inline conda/env/cli/__init__.py:1]: Oops, this empty submodule slipped through, removing it again

**Review by travishathaway — APPROVED** [review]

**Pull request merged**